code/game/machinery/computer/buildandrepair.dm

```diff
@@ -8,8 +8,21 @@
 	icon_state = "0"
 	var/state = 0
 	var/obj/item/weapon/circuitboard/circuit = null
+	var/computerframepasschance = 20
 //	weight = 1.0E8
 
+
+/obj/structure/computerframe/CanPass(atom/movable/mover, turf/target, height=0)
+	if(height==0)
+		return 1
+	if(istype(mover) && mover.checkpass(PASSGRILLE))
+		return prob(computerframepasschance)
+	else
+		if(istype(mover, /obj/item/projectile))
+			return prob(computerframepasschance)
+		else
+			return 0
+

```

Comment:
In this context it means a global proc. basically the stuff in the __HELPERS folder (see for instance game.dm), make a new proc at the end that encapsulates this stuff and have both girders and computer frames call that.